src/allmydata/test/test_istorageserver.py

```diff
@@ -1046,18 +1086,80 @@ def disconnect(self):
         assert self.storage_client is not current
 
 
+class _HTTPMixin(_SharedMixin):
+    """Run tests on the HTTP version of ``IStorageServer``."""
+
+    def setUp(self):
+        if PY2:
+            self.skipTest("Not going to bother supporting Python 2")
+        return _SharedMixin.setUp(self)
+
+    def _get_istorage_server(self):
+        set_treq_pool(HTTPConnectionPool(reactor, persistent=False))
+        swissnum = b"1234"
+        self._http_storage_server = HTTPServer(self.server, swissnum)
+        self._port_number = allocate_tcp_port()
+        self._listening_port = reactor.listenTCP(
+            self._port_number,
+            Site(self._http_storage_server.get_resource()),
+            interface="127.0.0.1",
+        )
+        return _HTTPStorageServer.from_http_client(
+            StorageClient(
+                DecodedURL.from_text("http://127.0.0.1:{}".format(self._port_number)),
```

Comment:
The hyperlink docs suggest:

```
DecodedURL().replace(scheme="http", host="127.0.0.1", port=1234)
```

I don't know why it doesn't just have `__init__` arguments ... but maybe that's still better than string generation and parsing.

Reply:
Done.